controllers/idler/idler_controller.go

```diff
@@ -244,18 +244,17 @@ func shorterDuration(first, second time.Duration) time.Duration {
 // Send notification if the deleted pod was managed by a controller, was a standalone pod that was not completed or was crashlooping
 func deletePodsAndCreateNotification(podCtx context.Context, pod corev1.Pod, r *Reconciler, idler *toolchainv1alpha1.Idler) error {
 	logger := log.FromContext(podCtx)
-	var podReason string
-	podCondition := pod.Status.Conditions
-	for _, podCond := range podCondition {
+	isCompleted := false
+	for _, podCond := range pod.Status.Conditions {
 		if podCond.Type == "Ready" {
-			podReason = podCond.Reason
+			isCompleted = podCond.Reason == "PodCompleted"
```

Comment:
We can break here, right?
```suggestion
			isCompleted = podCond.Reason == "PodCompleted"
			break
```

Reply:
of course :man_facepalming: 
thanks